x/ccv/provider/keeper/grpc_query.go

```diff
@@ -116,3 +120,109 @@ func (k Keeper) QueryValidatorProviderAddr(goCtx context.Context, req *types.Que
 		ProviderAddress: providerAddr.String(),
 	}, nil
 }
+
+// TODO: change to GlobalSlashEntries naming
+func (k Keeper) QueryPendingSlashPackets(goCtx context.Context, req *types.QueryPendingSlashPacketsRequest) (*types.QueryPendingSlashPacketsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	meter := k.GetSlashMeter(ctx)
+	allowance := k.GetSlashMeterAllowance(ctx)
+	lastTs := k.GetLastSlashMeterFullTime(ctx) // always UTC
+	nextReplenishTs := lastTs.Add(k.GetSlashMeterReplenishPeriod(ctx))
```

Comment:
Note that with new semantics, `LastSlashMeterFullTime` is the last time the slash meter was full, and the slash meter only gets replenished when it is not full. Ie. there are times where the slash meter will not be replenished for a long time (more than replenish period). 

Considering this, I think "next replenish" may be misleading if defined like this. Is it possible to make that a nullable value? If not, we can just return the last full time, along with the replenish period, and people can infer replenish times from those values.